yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -207,6 +211,17 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Gets the executor loss reason for a disconnected executor.
+   * Note that this method is expected to be called exactly once per executor ID.
+   */
+  def getExecutorLossReason(executorId: String): ExecutorLossReason = synchronized {
+    allocateResources()
```

Comment:
This is likely very noob of me so please pardon me, if so. Why do we need a call to allocateResources() here? I get processCompletedContainers() is important for you so the reasons get populated but wouldn't that have happened when the container in question was originally allocated by ApplicationMaster::launchReporterThread() (and subsequently killed).